internal/core/model.rs

```diff
@@ -1226,3 +1249,253 @@ fn test_vecmodel_set_vec() {
+            &self.notify
+        }
+    }
+
+    impl<M, F, T> ErasedRepeater for FilterProxy<M, F, T>
+    where
+        M: Model<Data = T> + 'static,
+        F: Fn(&T) -> bool + 'static,
+        T: 'static,
+    {
+        fn row_changed(&self, row: usize) {
+            let mut mapping = self.mapping.borrow_mut();
+
+            let index;
+            let is_contained = match mapping.binary_search(&row) {
+                Ok(i) => {
+                    index = i;
+                    true
+                }
+                Err(i) => {
+                    index = i;
+                    false
+                }
+            };
+
+            let should_be_contained =
+                (self.filter_function)(&self.inner_model.row_data(row).unwrap());
+
+            if is_contained && should_be_contained {
+                drop(mapping);
+                self.notify.row_changed(index);
+            } else if !is_contained && should_be_contained {
+                mapping.insert(index, row);
+                drop(mapping);
+                self.notify.row_added(index, 1);
+            } else if is_contained && !should_be_contained {
+                mapping.remove(index);
+                drop(mapping);
+                self.notify.row_removed(index, 1);
+            }
+        }
+
+        fn row_added(&self, index: usize, count: usize) {
+            if count == 0 {
+                return;
+            }
+
+            let insertion: Vec<usize> = self
+                .inner_model
+                .iter()
+                .enumerate()
+                .skip(index)
+                .take(count)
+                .filter_map(|(i, e)| if (self.filter_function)(&e) { Some(i) } else { None })
+                .collect();
+
+            if insertion.len() > 0 {
+                let mut mapping = self.mapping.borrow_mut();
+                let insertion_point = match mapping.binary_search(&index) {
+                    Ok(ip) => ip,
+                    Err(ip) => ip,
+                };
+
+                for (value, &count) in insertion.iter().enumerate() {
+                    mapping.insert(insertion_point + count, value);
+                }
+
+                drop(mapping);
+                self.notify.row_added(insertion_point, insertion.len());
+            }
+        }
+
+        fn row_removed(&self, index: usize, count: usize) {
+            if count == 0 {
+                return;
+            }
+            let mut mapping = self.mapping.borrow_mut();
+
+            let start = match mapping.binary_search(&index) {
+                Ok(i) => i,
+                Err(i) => i,
+            };
+            let end = match mapping.binary_search(&(index + count)) {
+                Ok(i) => i,
+                Err(i) => i,
+            };
+
+            let range = start..end;
+            if !range.is_empty() {
+                for _ in range.clone() {
+                    mapping.remove(start);
```

Comment:
again, you want to use Vec::splice.

... and fix all the values after end.

Reply:
I will do that as soon as everything works.
I think with https://doc.rust-lang.org/std/vec/struct.Vec.html#method.copy_within it is even better.